lib/rspec/matchers/built_in/compound.rb

```diff
@@ -101,6 +123,11 @@ def matcher_supports_block_expectations?(matcher)
           false
         end
 
+        def matcher_list_for(matcher)
```

Comment:
Same here...it'd be nice to have this make clear that it's just diffable matchers.  Maybe `diffable_matcher_list_from(matcher)`?